packages/components/tests/dummy/app/templates/components/alert.hbs

```diff
@@ -0,0 +1,257 @@
+{{page-title "Alert components"}}
+
+<h2 class="dummy-h2">Alerts</h2>
+
+<section>
+  <h3 class="dummy-h3">Scratch reference links</h3>
+  <p>Starting with Alert: "Global" and Alert: "Page or Inline". (Alert: "Toast" coming next.)</p>
+  <ul class="dummy-paragraph">
+    <li>
+      <a href="https://docs.google.com/document/d/1MC0MTm8-0F26s6Y-gou_RsMj6rtKS6jC-eM7rmruqLY/edit#heading=h.gjdgxs">
+        Alerts Components Requirement Document (CRD)
+      </a>
+    </li>
+    <li>
+      <a
+        href="https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/branch/Ru0h1oqy6r7SeGxwtQ6085/HDS-Product---Components-%5BWIP%5D?node-id=4023%3A15691"
+      >
+        WIP Figma file
+      </a>
+    </li>
+  </ul>
+</section>
+
+<section>
+  <h3 class="dummy-h3">Component API</h3>
+  <p class="dummy-paragraph">Here is the API for the component:</p>
+
+  <dl class="dummy-component-props">
+    <dt>type <code>enum</code></dt>
```

Comment:
[issue] personally I would not go with this approach (using a `prop` to choose which type of alert I want).

Main reason is that we can do the same using the name of the component itself – eg. `<Hds::Alert::Global>`, `<Hds::Alert::Page>`, `<Hds::Alert::Inline>`, `<Hds::Alert::Toast>` – to differentiate between components (they would share the same "base" implementation in code, using a "base" alert). This approach is more declarative and allows you to share the similarities but manage the differences, without squashing everything in a single implementation (eg. have the `isCompact` prop only for the "page or inline"). We can leverage composition more.

In general (and I have discussed this with @cveigt too, and looking at his Figma file seems it has followed the approach I suggested him) it's a "smell" when a component is described by its usage/context, instead that by its appearance (there are exceptions, of course: think of `LinkStandalone` which is a link used in a "standalone" context). If you look at the implementation in Figma, the "type" controls how it looks (`inline` vs `full-width`) and not where it's used.

Reply:
Makes sense now, put just the base here https://github.com/hashicorp/design-system/pull/92/commits/e4f4b7739c4a386e06a0bc2a7c5492fe57416251